src/main/java/org/springframework/integration/kafka/listener/AbstractOffsetManager.java

```diff
@@ -54,6 +54,8 @@
 
 	protected Map<Partition, Long> initialOffsets;
 
+	protected Map<Partition, Long> highestUpdatedOffsets = new ConcurrentHashMap<Partition, Long>();
```

Comment:
Can't we collapse this `*Offsets` Maps to the single one?
I don't `read` usage of the `highestUpdatedOffsets` hence I can simply update that `initialOffsets` for the new `highestUpdatedOffset` value.
WDYT ?